Sources/Sentry/SentryClient.m

```diff
@@ -253,6 +255,16 @@ - (SentryEvent *_Nullable)callEventProcessors:(SentryEvent *)event
     return newEvent;
 }
 
+- (void)setUserInfo:(NSDictionary *)userInfo withEvent:(SentryEvent *)event
+{
+    if (nil != event && nil != userInfo && userInfo.count > 0) {
+        if (nil == event.context) {
+            event.context = [[NSMutableDictionary alloc] init];
```

Comment:
`h`: Currently the `event.context` is always going to be `nil`, because we don't set it in `captureError` and `captureException`.  So we always create a mutable dictionary here. But if the `context` is not `nil` it also can be an immutable dictionary and we would crash when trying to set the `userInfo` below because we try to modify an immutable dictionary. 

So we could do something like this:
```
NSMutableDictionary *context;
if (nil == event.context) {
    context = [[NSMutableDictionary alloc] init];
} else {
    context = [event.context mutableCopy];
}

[context setValue:userInfo forKey:@"user info"];
event.context = context;
```

Or we just set override the context and reflect this in the functionName.
```
- (void)setContextFor:(SentryEvent *)event withUserInfo(NSDictionary *)userInfo
{
    if (nil != userInfo && userInfo.count > 0) {
        event.context = [[NSMutableDictionary alloc] init];
        [event.context setValue:userInfo forKey:@"user info"];
    }
}

```